Language Support for Java extension never finishes starting

Whenever the extension activates, its icon on the bottom-right starts "loading" and never turns into the "Thumbs Up" icon. Able to reproduce with individual Java files and Maven projects.  
Best problem indication so far is the console log "Error: Header must provide a Content-Length property".

The practical result is that the extension does not work at all (code navigation, intellisense, etc).

##### Environment
- Operating System: KDE neon (Linux hostname 5.4.0-72-generic 80-Ubuntu SMP Mon Apr 12 17:35:00 UTC 2021 x86_64 x86_64 x86_64 GNU/Linux)
- JDK version: OpenJDK jdk-11.0.10+9
- Visual Studio Code version:
    Version: 1.63.2
    Commit: 899d46d82c4c95423fb7e10e68eba52050e30ba3
    Date: 2021-12-15T09:39:46.686Z
    Electron: 13.5.2
    Chromium: 91.0.4472.164
    Node.js: 14.16.0
    V8: 9.1.269.39-electron.0
    OS: Linux x64 5.4.0-72-generic snap
- Java extension version: v1.2.0
- Other extensions installed:
  - Dart
  - Docker
  - Flutter
  - GitLens
  - JSON Tools
  - LaTeX Workshop
  - Vagrant
  - vscode-faker
  - XML Tools

##### Steps To Reproduce
1. Open a new VS Code window
2. Create a new Java file
3. The extension icon on the bottom-right starts "loading" and never turns into the "Thumbs Up" icon.

Also happens when I open a Maven project (my real-world use case).

##### What I've tried so far
- Removed `/home/username/.config/Code/User/workspaceStorage/` folder
- `F1` > `Java: Clean Java Language Server Workspace`
- Uninstalled Java extension pack and installed only the Language Support for Java extension
- Checked that I do *not* have the Visual Studio IntelliCode extension (according to [this issue](https://github.com/redhat-developer/vscode-java/issues/1040#issuecomment-524052548))

##### Logs

I set the `java.trace.server` setting to `verbose` and the extension log file got pretty big (1.4k lines), so I pasted it in pastebin (link below).

The VSCode's Developer Tools Console shows errors such as:
```
Error: Header must provide a Content-Length property.
	at m.onData (vscode-file://vscode-app/home/username/.vscode/extensions/redhat.java-1.2.0/dist/extension.js:2)
	at Socket.<anonymous> (vscode-file://vscode-app/home/username/.vscode/extensions/redhat.java-1.2.0/dist/extension.js:2)
	at Socket.emit (events.js:315)
	at addChunk (internal/streams/readable.js:309)
	at readableAddChunk (internal/streams/readable.js:284)
	at Socket.Readable.push (internal/streams/readable.js:223)
	at Pipe.onStreamRead (internal/stream_base_commons.js:188)
```


Java Extension log file: https://pastebin.com/nwL6SBS1

Java Language Server log file:
```
!SESSION 2022-01-20 11:02:57.307 -----------------------------------------------
eclipse.buildId=unknown
java.version=17.0.1
java.vendor=Eclipse Adoptium
BootLoader constants: OS=linux, ARCH=x86_64, WS=gtk, NL=en
Command-line arguments:  -data /home/username/.config/Code/User/workspaceStorage/1f338313d7b0a52013fc61c6213cc67e/redhat.java/jdt_ws

!ENTRY org.eclipse.jdt.ls.core 1 0 2022-01-20 11:02:58.449
!MESSAGE class org.eclipse.jdt.ls.core.internal.JavaLanguageServerPlugin is started

!ENTRY org.eclipse.jdt.ls.core 1 0 2022-01-20 11:02:58.615
!MESSAGE Main thread is waiting

!ENTRY org.eclipse.jdt.ls.core 1 0 2022-01-20 11:02:58.652
!MESSAGE >> initialize

!ENTRY org.eclipse.jdt.ls.core 1 0 2022-01-20 11:02:58.654
!MESSAGE Initializing Java Language Server 1.7.0.202112160206

!ENTRY org.eclipse.jdt.ls.core 1 0 2022-01-20 11:02:59.125
!MESSAGE Setting java.home /home/username/bin/jdk-11.0.10+9 as default global VM

!ENTRY org.eclipse.jdt.ls.core 1 0 2022-01-20 11:02:59.132
!MESSAGE Default VM Install changed from  1642687379113-/home/username/.vscode/extensions/redhat.java-1.2.0/jre/17.0.1-linux-x86_64 to 1642687379125-/home/username/bin/jdk-11.0.10+9

!ENTRY org.eclipse.jdt.ls.core 1 0 2022-01-20 11:02:59.142
!MESSAGE JVM Runtimes changed, saving new configuration

!ENTRY org.eclipse.jdt.ls.core 1 0 2022-01-20 11:02:59.146
!MESSAGE Static Commands: []

!ENTRY org.eclipse.jdt.ls.core 1 0 2022-01-20 11:02:59.147
!MESSAGE Non-Static Commands: [java.edit.organizeImports, java.project.refreshDiagnostics, java.project.import, java.navigate.openTypeHierarchy, java.project.removeFromSourcePath, java.project.listSourcePaths, java.project.resolveStackTraceLocation, java.project.getAll, java.project.isTestFile, java.project.getClasspaths, java.navigate.resolveTypeHierarchy, java.edit.stringFormatting, java.project.getSettings, java.project.updateSourceAttachment, java.project.resolveSourceAttachment, java.project.addToSourcePath]

!ENTRY org.eclipse.jdt.ls.core 1 0 2022-01-20 11:02:59.150
!MESSAGE Wait for AutoBuildOffJob start

!ENTRY org.eclipse.jdt.ls.core 1 0 2022-01-20 11:02:59.161
!MESSAGE Wait for AutoBuildOffJob end 11ms

!ENTRY org.eclipse.jdt.ls.core 1 0 2022-01-20 11:02:59.161
!MESSAGE Starting org.eclipse.m2e.core

!ENTRY org.eclipse.jdt.ls.core 1 0 2022-01-20 11:02:59.162
!MESSAGE Started org.eclipse.m2e.core 1ms

!ENTRY org.eclipse.jdt.ls.core 1 0 2022-01-20 11:02:59.162
!MESSAGE ProjectRegistryRefreshJob finished 0ms

!ENTRY org.eclipse.jdt.ls.core 1 0 2022-01-20 11:02:59.162
!MESSAGE Starting org.eclipse.buildship.core

!ENTRY org.eclipse.jdt.ls.core 1 0 2022-01-20 11:02:59.174
!MESSAGE Started org.eclipse.buildship.core 12ms

!ENTRY org.eclipse.jdt.ls.core 1 0 2022-01-20 11:02:59.185
!MESSAGE LoadingGradleVersionJob finished 11ms

!ENTRY org.eclipse.jdt.ls.core 1 0 2022-01-20 11:02:59.193
!MESSAGE Creating the Java project jdt.ls-java-project

!ENTRY org.eclipse.jdt.ls.core 1 0 2022-01-20 11:02:59.239
!MESSAGE Finished creating the Java project jdt.ls-java-project

!ENTRY org.eclipse.jdt.ls.core 1 0 2022-01-20 11:02:59.815
!MESSAGE Importing Maven project(s)

!ENTRY org.eclipse.jdt.ls.core 1 0 2022-01-20 11:03:03.539
!MESSAGE Workspace initialized in 4352ms

!ENTRY org.eclipse.jdt.ls.core 1 0 2022-01-20 11:07:08.799
!MESSAGE Parent process stopped running, forcing server exit

!ENTRY org.eclipse.jdt.ls.core 1 0 2022-01-20 11:07:08.800
!MESSAGE Shutdown received... waking up main thread

!ENTRY org.eclipse.jdt.ls.core 1 0 2022-01-20 11:07:08.818
!MESSAGE class org.eclipse.jdt.ls.core.internal.JavaLanguageServerPlugin is stopping:

!ENTRY org.eclipse.core.resources 2 10035 2022-01-20 11:07:09.431
!MESSAGE The workspace will exit with unsaved changes in this session.
!SESSION 2022-01-20 11:07:09.592 -----------------------------------------------
eclipse.buildId=unknown
java.version=17.0.1
java.vendor=Eclipse Adoptium
BootLoader constants: OS=linux, ARCH=x86_64, WS=gtk, NL=en
Command-line arguments:  -data /home/username/.config/Code/User/workspaceStorage/1f338313d7b0a52013fc61c6213cc67e/redhat.java/jdt_ws

!ENTRY org.eclipse.core.resources 2 10035 2022-01-20 11:07:10.251
!MESSAGE The workspace exited with unsaved changes in the previous session; refreshing workspace to recover changes.

!ENTRY org.eclipse.jdt.ls.core 1 0 2022-01-20 11:07:10.690
!MESSAGE class org.eclipse.jdt.ls.core.internal.JavaLanguageServerPlugin is started

!ENTRY org.eclipse.jdt.ls.core 1 0 2022-01-20 11:07:10.817
!MESSAGE Main thread is waiting

!ENTRY org.eclipse.jdt.ls.core 1 0 2022-01-20 11:07:10.842
!MESSAGE >> initialize

!ENTRY org.eclipse.jdt.ls.core 1 0 2022-01-20 11:07:10.843
!MESSAGE Initializing Java Language Server 1.7.0.202112160206

!ENTRY org.eclipse.jdt.ls.core 1 0 2022-01-20 11:07:10.859
!MESSAGE Static Commands: []

!ENTRY org.eclipse.jdt.ls.core 1 0 2022-01-20 11:07:10.860
!MESSAGE Non-Static Commands: [java.edit.organizeImports, java.project.refreshDiagnostics, java.project.import, java.navigate.openTypeHierarchy, java.project.removeFromSourcePath, java.project.listSourcePaths, java.project.resolveStackTraceLocation, java.project.getAll, java.project.isTestFile, java.project.getClasspaths, java.navigate.resolveTypeHierarchy, java.edit.stringFormatting, java.project.getSettings, java.project.updateSourceAttachment, java.project.resolveSourceAttachment, java.project.addToSourcePath]

!ENTRY org.eclipse.jdt.ls.core 1 0 2022-01-20 11:07:10.864
!MESSAGE Starting org.eclipse.m2e.core

!ENTRY org.eclipse.jdt.ls.core 1 0 2022-01-20 11:07:10.865
!MESSAGE Started org.eclipse.m2e.core 1ms

!ENTRY org.eclipse.jdt.ls.core 1 0 2022-01-20 11:07:11.574
!MESSAGE ProjectRegistryRefreshJob finished 709ms

!ENTRY org.eclipse.jdt.ls.core 1 0 2022-01-20 11:07:11.575
!MESSAGE Starting org.eclipse.buildship.core

!ENTRY org.eclipse.jdt.ls.core 1 0 2022-01-20 11:07:11.610
!MESSAGE Started org.eclipse.buildship.core 35ms

!ENTRY org.eclipse.jdt.ls.core 1 0 2022-01-20 11:07:11.625
!MESSAGE LoadingGradleVersionJob finished 14ms

!ENTRY org.eclipse.jdt.ls.core 1 0 2022-01-20 11:07:12.254
!MESSAGE Importing Maven project(s)

!ENTRY org.eclipse.jdt.ls.core 1 0 2022-01-20 11:07:12.369
!MESSAGE Workspace initialized in 743ms
```

Same issue.
Connecting to a docker container running in a remote host using ssh, testing with a HelloWorld.java, I find the Java Language Server never shows the "Thumbs Up" icon. The Java Build Status terminal outputs nothing.

The logs in Developer Tools:
```
workbench.desktop.main.js:2615 Error: Header must provide a Content-Length property.
	at p.onData (/root/.vscode-server/extensions/redhat.java-1.4.0-linux-x64/dist/extension.js:2:1001316)
	at Socket.<anonymous> (/root/.vscode-server/extensions/redhat.java-1.4.0-linux-x64/dist/extension.js:2:1001037)
	at Socket.emit (node:events:390:28)
	at addChunk (node:internal/streams/readable:315:12)
	at readableAddChunk (node:internal/streams/readable:289:9)
	at Socket.Readable.push (node:internal/streams/readable:228:10)
	at Pipe.onStreamRead (node:internal/stream_base_commons:199:23)
```

Java Language Server log:
```
!SESSION 2022-04-02 13:18:16.967 -----------------------------------------------
eclipse.buildId=unknown
java.version=17.0.2
java.vendor=Eclipse Adoptium
BootLoader constants: OS=linux, ARCH=x86_64, WS=gtk, NL=en
Command-line arguments:  -data /root/.vscode-server/data/User/workspaceStorage/96138e88cccd4c9222e51395fe593f1c/redhat.java/jdt_ws

!ENTRY org.eclipse.jdt.ls.core 1 0 2022-04-02 13:18:17.979
!MESSAGE class org.eclipse.jdt.ls.core.internal.JavaLanguageServerPlugin is started

!ENTRY org.eclipse.jdt.ls.core 1 0 2022-04-02 13:18:18.137
!MESSAGE Main thread is waiting

!ENTRY org.eclipse.jdt.ls.core 1 0 2022-04-02 13:18:18.158
!MESSAGE >> initialize

!ENTRY org.eclipse.jdt.ls.core 1 0 2022-04-02 13:18:18.160
!MESSAGE Initializing Java Language Server 1.9.0.202203030203

!ENTRY org.eclipse.jdt.ls.core 1 0 2022-04-02 13:18:18.191
!MESSAGE Setting java.home /usr/local/jdk1.8.0_321 as default global VM

!ENTRY org.eclipse.jdt.ls.core 1 0 2022-04-02 13:18:18.411
!MESSAGE Default VM Install changed from  1648905498182-/root/.vscode-server/extensions/redhat.java-1.4.0-linux-x64/jre/17.0.2-linux-x86_64 to 1648905498191-/usr/local/jdk1.8.0_321

!ENTRY org.eclipse.jdt.ls.core 1 0 2022-04-02 13:18:18.419
!MESSAGE JVM Runtimes changed, saving new configuration

!ENTRY org.eclipse.jdt.ls.core 1 0 2022-04-02 13:18:18.433
!MESSAGE Static Commands: []

!ENTRY org.eclipse.jdt.ls.core 1 0 2022-04-02 13:18:18.433
!MESSAGE Non-Static Commands: [java.edit.organizeImports, java.project.refreshDiagnostics, java.project.import, java.navigate.openTypeHierarchy, java.project.removeFromSourcePath, java.project.listSourcePaths, java.project.resolveStackTraceLocation, java.project.getAll, java.project.isTestFile, java.project.getClasspaths, java.navigate.resolveTypeHierarchy, java.edit.stringFormatting, java.project.getSettings, java.project.updateSourceAttachment, java.project.upgradeGradle, java.project.resolveSourceAttachment, java.project.addToSourcePath]

!ENTRY org.eclipse.jdt.ls.core 1 0 2022-04-02 13:18:18.434
!MESSAGE Static Commands: []

!ENTRY org.eclipse.jdt.ls.core 1 0 2022-04-02 13:18:18.434
!MESSAGE Non-Static Commands: [java.maven.initializeSearcher, java.maven.searchArtifact, java.maven.addDependency, java.maven.controlContext]

!ENTRY org.eclipse.jdt.ls.core 1 0 2022-04-02 13:18:18.434
!MESSAGE Static Commands: []

!ENTRY org.eclipse.jdt.ls.core 1 0 2022-04-02 13:18:18.434
!MESSAGE Non-Static Commands: [vscode.java.test.findJavaProjects, vscode.java.test.findTestPackagesAndTypes, vscode.java.test.findTestTypesAndMethods, vscode.java.test.resolvePath, vscode.java.test.findTestLocation, vscode.java.test.get.testpath, vscode.java.test.findDirectTestChildrenForClass, vscode.java.test.navigateToTestOrTarget, vscode.java.test.junit.argument, vscode.java.test.generateTests]

!ENTRY org.eclipse.jdt.ls.core 1 0 2022-04-02 13:18:18.434
!MESSAGE Static Commands: []

!ENTRY org.eclipse.jdt.ls.core 1 0 2022-04-02 13:18:18.434
!MESSAGE Non-Static Commands: [java.project.refreshLib, java.project.list, java.project.generateJar, java.project.getMainClasses, java.getPackageData, java.resolvePath]

!ENTRY org.eclipse.jdt.ls.core 1 0 2022-04-02 13:18:18.435
!MESSAGE Static Commands: []

!ENTRY org.eclipse.jdt.ls.core 1 0 2022-04-02 13:18:18.435
!MESSAGE Non-Static Commands: [vscode.java.checkProjectSettings, vscode.java.isOnClasspath, vscode.java.fetchUsageData, vscode.java.validateLaunchConfig, vscode.java.resolveInlineVariables, vscode.java.resolveClassFilters, vscode.java.resolveMainMethod, vscode.java.resolveClasspath, vscode.java.resolveBuildFiles, vscode.java.resolveMainClass, vscode.java.updateDebugSettings, vscode.java.resolveSourceUri, vscode.java.fetchPlatformSettings, vscode.java.buildWorkspace, vscode.java.startDebugSession, vscode.java.inferLaunchCommandLength, vscode.java.resolveElementAtSelection, vscode.java.resolveJavaExecutable]

!ENTRY org.eclipse.jdt.ls.core 1 0 2022-04-02 13:18:18.435
!MESSAGE Static Commands: []

!ENTRY org.eclipse.jdt.ls.core 1 0 2022-04-02 13:18:18.435
!MESSAGE Non-Static Commands: [java.intellicode.completion, java.intellicode.completion.resolve, java.intellicode.enable]

!ENTRY org.eclipse.jdt.ls.core 1 0 2022-04-02 13:18:18.440
!MESSAGE Wait for AutoBuildOffJob start

!ENTRY org.eclipse.jdt.ls.core 1 0 2022-04-02 13:18:18.451
!MESSAGE Wait for AutoBuildOffJob end 11ms

!ENTRY org.eclipse.jdt.ls.core 1 0 2022-04-02 13:18:18.452
!MESSAGE Starting org.eclipse.m2e.core

!ENTRY org.eclipse.jdt.ls.core 1 0 2022-04-02 13:18:18.452
!MESSAGE Started org.eclipse.m2e.core 0ms

!ENTRY org.eclipse.jdt.ls.core 1 0 2022-04-02 13:18:18.452
!MESSAGE ProjectRegistryRefreshJob finished 0ms

!ENTRY org.eclipse.jdt.ls.core 1 0 2022-04-02 13:18:18.452
!MESSAGE Starting org.eclipse.buildship.core

!ENTRY org.eclipse.jdt.ls.core 1 0 2022-04-02 13:18:18.504
!MESSAGE Started org.eclipse.buildship.core 51ms

!ENTRY org.eclipse.jdt.ls.core 1 0 2022-04-02 13:18:18.514
!MESSAGE LoadingGradleVersionJob finished 10ms

!ENTRY org.eclipse.jdt.ls.core 1 0 2022-04-02 13:18:18.529
!MESSAGE Creating the Java project jdt.ls-java-project

!ENTRY org.eclipse.jdt.ls.core 1 0 2022-04-02 13:18:18.584
!MESSAGE Finished creating the Java project jdt.ls-java-project

!ENTRY org.eclipse.jdt.ls.core 1 0 2022-04-02 13:18:19.037
!MESSAGE Try to create an invisible project for the workspace /root/hello_java

!ENTRY org.eclipse.jdt.ls.core 1 0 2022-04-02 13:18:19.037
!MESSAGE Creating the Java project hello_java_3dbf08ad

!ENTRY org.eclipse.jdt.ls.core 1 0 2022-04-02 13:18:19.076
!MESSAGE Finished creating the Java project hello_java_3dbf08ad

!ENTRY org.eclipse.jdt.ls.core 1 0 2022-04-02 13:18:19.085
!MESSAGE Successfully created a workspace invisible project hello_java_3dbf08ad

!ENTRY org.eclipse.jdt.ls.core 1 0 2022-04-02 13:18:19.119
!MESSAGE Workspace initialized in 604ms

!ENTRY org.eclipse.jdt.ls.core 1 0 2022-04-02 13:18:20.167
!MESSAGE >> Updating classpath for project hello_java_3dbf08ad

```

When I connect to the host rather than the container, testing with the same java file, everything works fine.

Env in container:
OS: Linux 4488f5b72f2d 4.15.0-76-generic #86-Ubuntu SMP Fri Jan 17 17:24:28 UTC 2020 x86_64 x86_64 x86_64 GNU/Linux
JDK: jdk1.8.0_321
Java Extension: 1.4.0 (March 3rd, 2022)

--- a/package.json
+++ b/package.json
@@ -213,8 +213,8 @@
             "string",
             "null"
           ],
-          "default": "-XX:+UseParallelGC -XX:GCTimeRatio=4 -XX:AdaptiveSizePolicyWeight=90 -Dsun.zip.disableMemoryMapping=true -Xmx1G -Xms100m -Xlog:jni+resolve=off",
-          "description": "Specifies extra VM arguments used to launch the Java Language Server. Eg. use `-XX:+UseParallelGC -XX:GCTimeRatio=4 -XX:AdaptiveSizePolicyWeight=90 -Dsun.zip.disableMemoryMapping=true -Xmx1G -Xms100m -Xlog:jni+resolve=off` to optimize memory usage with the parallel garbage collector",
+          "default": "-XX:+UseParallelGC -XX:GCTimeRatio=4 -XX:AdaptiveSizePolicyWeight=90 -Dsun.zip.disableMemoryMapping=true -Xmx1G -Xms100m -Xlog:disable",
+          "description": "Specifies extra VM arguments used to launch the Java Language Server. Eg. use `-XX:+UseParallelGC -XX:GCTimeRatio=4 -XX:AdaptiveSizePolicyWeight=90 -Dsun.zip.disableMemoryMapping=true -Xmx1G -Xms100m -Xlog:disable` to optimize memory usage with the parallel garbage collector",
           "scope": "machine-overridable"
         },
         "java.errors.incompleteClasspath.severity": {

--- a/src/javaServerStarter.ts
+++ b/src/javaServerStarter.ts
@@ -112,8 +112,8 @@ function prepareParams(requirements: RequirementsData, javaConfiguration, worksp
 			params.push(`${watchParentProcess}false`);
 		}
 	}
-	if (vmargs.indexOf('-Xlog:jni+resolve=') < 0) {
-		params.push('-Xlog:jni+resolve=off');
+	if (vmargs.indexOf('-Xlog:') < 0) {
+		params.push('-Xlog:disable');
 	}
 
 	parseVMargs(params, vmargs);

